modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -0,0 +1,671 @@
+			$icon = blavatar_url( $domain, 'img', 32 );
+		}
+
+		$badge = '';
+		if ( $this->is_wpcom_site() ) {
+			if ( function_exists( 'is_private_blog' ) && function_exists( 'wpcom_is_coming_soon' ) && is_private_blog() ) {
+				$badge .= sprintf(
+					'<span class="site__badge site__badge-private">%s</span>',
+					wpcom_is_coming_soon() ? esc_html__( 'Coming Soon', 'jetpack' ) : esc_html__( 'Private', 'jetpack' )
+				);
+			}
+
+			if ( function_exists( 'is_redirected_domain' ) && is_redirected_domain( $domain ) ) {
+				$badge .= '<span class="site__badge site__badge-redirect">' . esc_html__( 'Redirect', 'jetpack' ) . '</span>';
+			}
+
+			if ( ! empty( get_option( 'options' )['is_domain_only'] ) ) {
+				$badge .= '<span class="site__badge site__badge-domain-only">' . esc_html__( 'Domain', 'jetpack' ) . '</span>';
+			}
+		}
+
+		if ( jetpack_is_atomic_site() ) {
+			if ( function_exists( 'site_is_private' ) && function_exists( 'site_is_coming_soon' ) && site_is_private() ) {
+				$badge .= sprintf(
+					'<span class="site__badge site__badge-private">%s</span>',
+					site_is_coming_soon() ? esc_html__( 'Coming Soon', 'jetpack' ) : esc_html__( 'Private', 'jetpack' )
+				);
+			}
+		}
+
+		$site_card = '
+<div class="site__info">
+	<div class="site__title">%1$s</div>
+	<div class="site__domain">%2$s</div>
+	%3$s
+</div>';
+
+		$site_card = sprintf(
+			$site_card,
+			get_option( 'blogname' ),
+			$domain,
+			$badge
+		);
+
+		add_menu_page( 'site-card', $site_card, 'read', get_home_url(), null, $icon, 1 );
+		add_filter( 'add_menu_classes', array( $this, 'set_site_card_menu_class' ) );
+	}
+
+	/**
+	 * Adds a custom element class and id for Site Card's menu item.
+	 *
+	 * @param array $menu Associative array of administration menu items.
+	 * @return array
+	 */
+	public function set_site_card_menu_class( array $menu ) {
+		foreach ( $menu as $key => $menu_item ) {
+			if ( 'site-card' !== $menu_item[3] ) {
+				continue;
+			}
+
+			$classes = ' toplevel_page_site-card';
+
+			// webclip.png is the default on WoA sites. Anything other than that means we have a custom site icon.
+			if (
+				( function_exists( 'blavatar_exists' ) && blavatar_exists( ( new Status() )->get_site_suffix() ) ) ||
+				( has_site_icon() && 'https://s0.wp.com/i/webclip.png' !== get_site_icon_url( 512 ) )
+			) {
+				$classes .= ' has-site-icon';
+			}
+
+			$menu[ $key ][4] = $menu_item[4] . $classes;
+			$menu[ $key ][5] = 'toplevel_page_site_card';
+			break;
+		}
+
+		return $menu;
+	}
+
+	/**
+	 * Adds My Home menu.
+	 *
+	 * @param string $domain  Site domain.
+	 * @param bool   $calypso Optional. Whether links should point to Calypso or wp-admin. Default true (Calypso).
+	 */
+	public function add_my_home_menu( $domain, $calypso = true ) {
+		global $submenu;
+
+		$menu_slug = $calypso ? 'https://wordpress.com/home/' . $domain : 'index.php';
```

Comment:
I think we'll want to use `Automattic\Jetpack\Redirect` for all the links, just like we do in other places of Jetpack. This gives us more control, and allows things like remote updates in the future as well as tracking (which is done by Tracks in the Masterbar drawers until now).

Reply:
I started to work on this, and while it worked as expected in wp-admin, it didn't really work for API version of it that gets consumed by Calypso. How strongly do you feel about this?

Reply:
I don't have a strong opinion about this. The Redirect is useful if:

- you intend on using the data it would gather.
- you think we may change where some of those links point to in the future.

If that doesn't seem useful to you, I suppose it can stay the way it is today.